mage_ai/data_preparation/models/file.py

```diff
@@ -54,10 +43,45 @@ def delete(self):
         os.remove(self.file_path)
 
     def to_dict(self, include_content=False):
-        data = dict(
-            name=self.filename,
-            path=os.path.join(self.dir_path, self.filename)
-        )
+        data = dict(name=self.filename, path=os.path.join(self.dir_path, self.filename))
         if include_content:
             data['content'] = self.content()
         return data
+
+
+def traverse(directory: str) -> Dict:
+    """
+    Traverses directory tree and returns the underlying file structure
+
+    Args:
+        directory (str): Top-level directory to generate tree from
+
+    Returns:
+        Dict: Directory tree payload representation
+    """
+    name = os.path.splitext(os.path.basename(directory))[0]
+    disabled = name[0] == '.' or name in INACCESSIBLE_DIRS
+    children = []
+    entries = os.scandir(directory)
+    children.extend(
+        _traverse(entry, disabled, 1) for entry in sorted(entries, key=lambda entry: entry.name)
+    )
+    return dict(name=name, children=children)
+
+
+def _traverse(entry, disabled=True, depth=1) -> Dict:
```

Comment:
How are you using depth? Do you want to stop traverse when depth > MAX_DEPTH?

Reply:
A check against max depth accidentally got deleted at some point, I added it back. I stop traversal when `depth > MAX_DEPTH`.